app/Traits/FilterTrait.php

```diff
@@ -0,0 +1,72 @@
+<?php
+namespace App\Traits;
+
+use Illuminate\Support\Facades\Schema;
+
+trait FilterTrait
+{
+    /**
+     * Filter the result follow the filter request.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query  of Model.
+     * @param array                                 $fields fields
+     *
+     * @return void.
+     */
+    public function scopeFilter($query, $fields)
+    {
+        if (isset($fields)) {
+            foreach ($fields as $field => $value) {
+                if ($this->getColummnsFilter($field) !=null) {
```

Comment:
Lưu `$this->getColummnsFilter($field)` vào 1 biến. tránh trường hợp gọi function này 2 lần.